src/nodes/chunk_insert_state.c

```diff
@@ -585,12 +639,27 @@ ts_chunk_insert_state_create(const Chunk *chunk, ChunkDispatch *dispatch)
 		resrelinfo->ri_IndexRelationDescs == NULL)
 		ExecOpenIndices(resrelinfo, onconflict_action != ONCONFLICT_NONE);
 
-	if (resrelinfo->ri_TrigDesc != NULL)
+	if (relinfo->ri_TrigDesc != NULL)
 	{
-		if (resrelinfo->ri_TrigDesc->trig_insert_instead_row ||
-			resrelinfo->ri_TrigDesc->trig_insert_after_statement ||
-			resrelinfo->ri_TrigDesc->trig_insert_before_statement)
-			elog(ERROR, "insert trigger on chunk table not supported");
+		TriggerDesc *tg = relinfo->ri_TrigDesc;
+
+		/* instead of triggers can only be created on VIEWs */
+		Assert(!tg->trig_insert_instead_row);
+
+		/*
+		 * A statement that targets a parent table in an inheritance or
+		 * partitioning hierarchy does not cause the statement-level triggers
+		 * of affected child tables to be fired; only the parent table's
+		 * statement-level triggers are fired. However, row-level triggers
+		 * of any affected child tables will be fired.
+		 * During chunk creation we only copy ROW trigger to chunks so
+		 * statement triggers should not exist on chunks.
+		 */
+		if (tg->trig_insert_after_statement || tg->trig_insert_before_statement)
```

Comment:
Don't think process_utility blocks things at the chunk level. Only at the hypertable level. Safer to have this throw an error here.

Reply:
But the comment above explains that statement triggers are not fired on inheritance children. Since this is part of the chunk routing code, the insert must be via the inheritance parent. So, either the comment above is wrong or this error can never happen.

Reply:
A user can still  manually create stmt triggers on the chunks. Nothing that blocks it. For compressed chunks, we have to do the trigger execution manually before redirecting the tuple to the internal compressed chunk.  This is not automatically handled by PG code.

Reply:
But this path is for inserts via the root hypertable. For direct inserts on chunks, this code is, AFAIK, not executed.